homeassistant/components/mqtt/__init__.py

```
@@ -248,12 +260,13 @@ def remove():
 
 
 @asyncio.coroutine
-def _async_setup_server(hass, config):
+def _async_setup_server(hass: HomeAssistantType,
+                        config: ConfigType):
```

Returns None to indicate a failed state instead of False+None.